src/controllers/wishlist.controller.js

```diff
@@ -8,37 +8,40 @@ const addWishlist=async(req, res)=> {
     try {
         const token=req.header('Authorization').split(' ')[1]
         const userVerify=verfyToken(token,process.env.JWT_SECRET)
+        
         if (!req.user) {
             return res.status(401).json({ message: "unauthorized user!" });
         }
-        const { userId, productId } = req.body;
+       
+        const { productId } = req.body;
         const product = await Products.findByPk(productId);
-         const user = await User.findByPk(userId);
 
         if (!product) {
            return res.status(400).json({ message: "product not found" });
         }
-         if (userVerify.data.id !== user.id) {
+         if (!userVerify.data.id) {
              return res.status(401).json({ message: 'not permited!' });
            }
            
-        const wishes = await ProductWish.findOne({ where: { userId, productId } });
+        const wishes = await ProductWish.findOne({ where: { productId,userId:userVerify.data.id } });
         if (wishes) {
-            console.log(wishes)
             await wishes.destroy();
             return res.status(200).json({ message: "product unwished  succesfully!" });
         }
-        
+        if(userVerify.data.role==="buyer"){
         const wishedItem = {
-            userId: user.id,
+            userId: userVerify.data.id,
             productId: product.id
         };
-        console.log(wishedItem)
         await createAwish(wishedItem);
         return res.status(201).json({
             message: 'product wished created successfull',
             wishedItemCreate: wishedItem,
         });
+    }
+        else{
+            return res.status(200).json({message:"only buyer are allowed!"})
```

Comment:
Mhn try to make your functions as short as possible

Reply:
done